### ISSUE TITLE
Feature request: I see only 8-bit is supported. Is there a plan to support 10, 12-bit? 

### ISSUE THREAD

**Issue by nitheeshkm** [KCL-BMEIS/pymagewell#7]

**Comment by crnbaker** [comment]
Sorry for the slow reply. I'd be up for that - is it something all cards support?

**Comment by nitheeshkm** [comment]
Thanks for replay. I'm not sure if all cards support but anything with plus support 10-bit.

I'm using this:
https://www.magewell.com/products/pro-capture-sdi-4k-plus

**Comment by BSIMike** [comment]
10bit would be awesome! That opens up all the new HDR / UHD formats!

**Comment by crnbaker** [comment]
It's been a while since I've worked on this, but I've had a quick look and here's what I've found out (notes here for myself when I can pick it up or anyone else who wants to make a fork and try to implement themselves!):
* You can already choose 10-bit formats from the `ColourFormat` enum when you create your settings object; Y410 for example:
```
device_settings = ProCaptureSettings(
    dimensions=ImageSizeInPixels(1920, 1080),
    color_format=ColourFormat.Y410, # Color format of captured video frames
    transfer_mode=TransferMode.LOW_LATENCY,
    num_lines_per_chunk=64  # has effect only in low latency mode
)
```
* The formats available in the enum are taken from the SDK, so presumably include all formats support by the various ProCapture cards
* Currently, the only way to access a grabbed frame is to convert it from its string_buffer format to a numpy array with `VideoFrame.as_array()`. This method currently assumes that the frame is in an 8-bit BGR format (e.g. BGR24).

So I think we need to add something to `VideoFrame.as_array()` to detect the `ColourFormat` and properly decode it. Maybe there's something available in opencv, which would be good because it's already a dependency. Or we could use ffmpeg. Specific details on the YUV formats (e.g. Y410) are [here](https://learn.microsoft.com/en-us/windows/win32/medfound/10-bit-and-16-bit-yuv-video-formats).

Contributions welcome!

**Comment by nitheeshkm** [comment]
@crnbaker, I think this is what we need. We can use the functions to generate NumPy array and then do the rest of the processing ourselves. ffpmpeg integration into pymagewell isn't something we are looking for as we have our own ffmpeg commands. 

Thank you for implementing this. I will test it in near future and close the thread.

**Comment by crnbaker** [comment]
Great thanks! I haven't pushed it yet, I'll make a PR at some point soon and add ask you to have a look at the branch before merging if you don't mind.

**Comment by nitheeshkm** [comment]
That sounds good!

**Comment by crnbaker** [comment]
@nitheeshkm take a look at the [PR](https://github.com/KCL-BMEIS/pymagewell/pull/10). I'm sure you know this already but you should be able to install the branch to your environment with `pip install git+https://github.com/tangentlabs/django-oscar-paypal.git@7_rgb_video_decoding`. Thanks!

**Comment by crnbaker** [comment]
Feature available in v0.2.0

source: https://github.com/KCL-BMEIS/pymagewell/tree/v0.2.0
release: https://github.com/KCL-BMEIS/pymagewell/releases/tag/v0.2.0
